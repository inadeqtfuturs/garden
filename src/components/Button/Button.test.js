import React from 'react';
import { Button } from '@components';
import ThemeRender from '@utils/testing';

describe('Button', () => {
  it('renders correctly', () => {
    const tree = ThemeRender(<Button>Test</Button>);
    expect(tree).toMatchSnapshot();
  });

  it('takes props', () => {
    const tree = ThemeRender(
      <Button color="secondary" size="lg">
        Secondary Button
      </Button>
    );
    expect(tree).toMatchSnapshot();
  });

  it('can be disabled', () => {
    const tree = ThemeRender(<Button disabled>disabled button</Button>);
    expect(tree).toMatchSnapshot();
  });
});
