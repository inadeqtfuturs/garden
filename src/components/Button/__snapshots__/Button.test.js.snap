// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button can be disabled 1`] = `
<button
  className="Button__StyledButton-sc-rouxfp-0 dFIsDi"
  color="primary"
  disabled={true}
  onClick={[Function]}
  size="md"
  type="default"
>
  disabled button
</button>
`;

exports[`Button renders correctly 1`] = `
<button
  className="Button__StyledButton-sc-rouxfp-0 kzTaGv"
  color="primary"
  disabled={false}
  onClick={[Function]}
  size="md"
  type="default"
>
  Test
</button>
`;

exports[`Button takes props 1`] = `
<button
  className="Button__StyledButton-sc-rouxfp-0 juTlWk"
  color="secondary"
  disabled={false}
  onClick={[Function]}
  size="lg"
  type="default"
>
  Secondary Button
</button>
`;
